dm/dm/config/subtask.go

```diff
@@ -472,6 +473,18 @@ func (c *SubTaskConfig) Adjust(verifyDecryptPassword bool) error {
 		return terror.ErrConfigBinlogEventFilter.Delegate(err)
 	}
 
+	// adjust s3 dir
+	if c.Mode == ModeAll || c.Mode == ModeFull {
+		isS3, newDir, err := s3.AdjustS3Path(c.LoaderConfig.Dir, c.SourceID)
```

Comment:
task name is also needed to prevent overwrite

Reply:
yes, but task name has been add at [L430](https://github.com/pingcap/tiflow/pull/4361/files#diff-65dffe267f5514eeb5de94bac58d0944a395087f6081abcfdeafca754935862fL429)

Reply:
that adding logic can't append task name to a s3 URL with query parameters. please fix it

Reply:
ok, i understand.